### DISCUSSION TITLE
ApiService with custom Kestrel/HttpsFromPem config

### DISCUSSION THREAD

**Discussion by fergalmoran** [microsoft/aspire#3319] [Q&A] [answered]
I have this in my launchsettings.json for my ApiService and it starts up fine (and serves the correct cert) when I run my project from the AppHost folder.

```  
"Kestrel": {
    "Endpoints": {
      "HttpsFromPem": {
        "Certificate": {
          "Path": "/etc/letsencrypt/live/<mylocaldomain>/cert.pem",
          "KeyPath": "/etc/letsencrypt/live/<mylocaldomain>/privkey.pem"
        },
        "Protocols": "Http1AndHttp2AndHttp3",
        "Url": "https://<mylocaldomain>:5001"
      }
    }
  }
```

However - the Endpoints column isn't populated properly in the dashboard?
![image](https://github.com/dotnet/aspire/assets/309139/215eb540-9d23-4119-9923-f2c67a50b004)

What am I missing?

**Comment by davidfowl** [comment, marked as answer]
This is a reasonable feature request. We don't read native kestrel configuration and hoist them into endpoints in the app model. We could do that.

**Reply by fergalmoran** [reply]
@davidfowl should I convert this to an issue or just leave it with ye?

**Reply by davidfowl** [reply]
Already did https://github.com/dotnet/aspire/issues/3456
